packages/itwinui-css/src/dialog/dialog.scss

```diff
@@ -51,11 +51,48 @@
   }
 }
 
-@mixin iui-dialog-default {
-  .iui-dialog {
-    left: 50%;
+@mixin iui-dialog-position {
+  top: 33%;
+  left: 50%;
+  transform: translate(-50%, -33%);
+
+  &[data-position] {
+    position: fixed;
+    top: auto;
+    left: auto;
+    transform: none;
+  }
+
+  &[data-position='centered'] {
+    position: fixed;
     top: 33%;
+    left: 50%;
     transform: translate(-50%, -33%);
+  }
+
+  &[data-position='top-left'] {
+    top: $iui-baseline;
+    left: $iui-m;
```

Comment:
I think it would make more sense to use `top: 0`, `left: 0`, etc and then apply margin